### ISSUE TITLE
Compile error with 'debug-embed'

### ISSUE THREAD

**Issue by jangernert** [pyros2097/rust-embed#50]
I have a base library which embeds some icons using `rust-embed`. When using the library from another project the embedded resources can only be found when running the binary in release mode. Trying to run the application in debug mode where the resources are read from the file system yields run-time errors.
So I would like to turn on the `debug-embed` feature to always embed the resource like in the release build.
However by doing this
```
rust-embed = { version = "4.2", features = ["debug-embed"] }
```

I get the following compile errors
```
  |
9 | #[derive(RustEmbed)]
  |          ^^^^^^^^^ could not find `utils` in `rust_embed`
```

Any help would be greatly appreciated.

**Comment by pyros2097** [comment]
Hmmm. I guess we need a test case for this. Is Anyone here who worked on this feature can take a look at this?

**Comment by jangernert** [comment]
I was just about to set up a minimal sample of my use case when I noticed, I don't even need to have two different crates.
A bare bone library crate with only `rust-embed` with enabled `debug-embed` feature as dependency is enough to yield the exact same compilation error.

```
[package]
name = "base_lib"
version = "0.1.0"
authors = ["Jan Lukas Gernert <jangernert@gmail.com>"]
edition = "2018"

[dependencies]
rust-embed = { version = "4.2", features = ["debug-embed"] }
```

```
use rust_embed::RustEmbed;

#[derive(RustEmbed)]
#[folder = "resources"]
struct Resources;

#[cfg(test)]
mod tests {
    #[test]
    fn it_works() {
        assert_eq!(2 + 2, 4);
    }
}
```

**Comment by pyros2097** [comment]
Yeah thats what i was thinking this it could be that we never tested this feature but merged it. And its time we testing this. I think @lukad  was the one who implemented it. I guess we forgot to test it out.

**Comment by lukad** [comment]
Looks like this was broken in !45. The checks for conditional compilation https://github.com/vemoo/rust-embed/blob/5076774942db5ecfb5493f4f8183c14159975ada/src/lib.rs#L1 and https://github.com/vemoo/rust-embed/blob/5076774942db5ecfb5493f4f8183c14159975ada/src/lib.rs#L10 don't include walkdir and utils when building with debug-embed.
I'm not sure what the correct check should be but I'll take another look at this when I'm home.

**Comment by lukad** [comment]
And I agree, there should be a test for this.

**Comment by lukad** [comment]
To be hones I have absolutely no clue how the current code works.
I tried removing the `not(feature = "debug-embed")` check from the lines mentioned above but I don't know if that's the correct way. @vemoo Can you please take a look at this?

**Comment by vemoo** [comment]
The issue was that after splitting it in two crates the feature has to be passed to the impl create.
I've also added a step to test that in the CI.

**Comment by jangernert** [comment]
4.3 is working great for me. Thx
